### PR TITLE
[Feature] Notify Tighten When Resources Expire

### LINKED ISSUE #406: Notify Tighten of expired content
Once a week, we should send a message to the ops channel in Slack with a compiled list of all of the expired resources. See #391 for additional context.

### PR DESCRIPTION
**Pull request by TammyTee**
## Summary

This PR adds a `resource:expired --notify` command scheduled to run weekly on Fridays at 6:00AM. When there are expired resources or resources expiring within 2 weeks' time, a detailed report is sent to slack:

image

When the command is run without the `--notify` option, an output of expired resources are printed in the terminal:

image

### `Resource::class` Updates

- added a `days_til_expired` attribute
- added an `isExpired` and `isExpiring` helper method
- added two new query scopes: `scopeExpired` and `scopeExpiring` - the latter will return resources expiring within the next two weeks

### Other
- a new migration was added to allow resources where `can_expire` is `false` to have a nullable `expiration_date`

Resolves #406

### CODE REVIEW

**Comment by TammyTee** [comment]
Currently, all content will have an expiration 6 months from the date it was added. I wonder if we should consider different defaults based on the resource type e.g. course and video resources may change content/urls more often so we should check that those are good more frequently. Thoughts?

File: app/Console/Commands/GetExpiredResources.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            ->orWhere(function ($query) {
+                $query->expiring();
+            })
+            ->with('modules')
+            ->get([
+                'name',
+                'url',
+                'expiration_date',
+                'created_at',
+            ]);
```
Comment (darkboywonder) [inline app/Console/Commands/GetExpiredResources.php:27]: OH this is new to me 🍿 . That's 🔥

**Review by darkboywonder — APPROVED** [review]

**Comment by darkboywonder** [comment]
> Currently, all content will have an expiration 6 months from the date it was added. I wonder if we should consider different defaults based on the resource type e.g. course and video resources may change content/urls more often so we should check that those are good more frequently. Thoughts? @darkboywonder @MarcyLina

🌞 yooooooooooooooo
Since this is new ground, I think we should stick with the default six months for everything and slowly adjust default dates for sub-categories that show a repeating pattern. I feel we might even add more sub-categories or tags that link to framework specifics. Then, we can set default expiration dates based on a more solid notion than the current categories, which are generic and don't genuinely relate to when they should expire. An example of this would be the content we link to in the Laravel documentation. We know those will need a new link roughly every six months, so we might want to add a sub-category tag like "Laravel" that we base our default on instead of just "Article" or "Documentation."

**Comment by TammyTee** [comment]
> Since this is new ground, I think we should stick with the default six months for everything and slowly adjust default dates for sub-categories that show a repeating pattern.

@darkboywonder totally agree! I didn't think about this. Thanks for chiming in 😄

**Comment by MarcyLina** [comment]
A very belated "I agree with Shawn" lol

**Pull request merged**